components/notebook-controller/config/overlays/openshift/generate-metadata-yaml.sh

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env bash
+
+set -uo pipefail
+
+function trap_exit() {
+	rc=$?
+
+	set +x
+
+	exit $rc
+}
+
+trap "trap_exit" EXIT
+
+_derive_metadata()
+{
+  # inspired from https://stackoverflow.com/a/29835459
+  current_dir=$(CDPATH= cd -- "$(dirname -- "$0")" && pwd)
+
+  kf_project_file="${current_dir}/PROJECT"
```

Comment:
**`"Controversial" Decision #1`**

I noticed most kubeflow components have this `PROJECT` file in the component root directory that contains (loosely) the information required by the `odh_metadata.yaml` _specification_.

I decided to fully embrace auto-generating this file if the present of a `PROJECT` file was found (while still supporting command line flag overrides to explicitly provide the value.

Good idea? Bad Idea? _You tell me!_